integrationtest/wildfly/src/test/java/org/hibernate/search/test/integration/jms/transaction/TransactionalJmsMasterSlave.java

```diff
@@ -181,42 +184,21 @@ public void searchRollbackedMemberAfterSynchronizationOnSlave2() throws Exceptio
 	@OperateOnDeployment("slave-2")
 	public void searchNameShouldWorkOnSlave2() throws Exception {
 		memberRegistration.assertConfiguration( "Test Sequence 9", "slave-2", JndiJMSBackendQueueProcessor.class.getName() );
-		assertEquals( "Davide D'Alto", findAtLeastOneName( "davide" ).get( 0 ) );
-		assertEquals( "Peter O'Tall", findAtLeastOneName( "peter" ).get( 0 ) );
-		assertEquals( "Richard Mayhew", findAtLeastOneName( "richard" ).get( 0 ) );
+		POLLER.pollAssertion( () -> assertAtLeastOneName( "Davide D'Alto", "davide" ) );
+		POLLER.pollAssertion( () -> assertAtLeastOneName( "Peter O'Tall", "peter" ) );
+		POLLER.pollAssertion( () -> assertAtLeastOneName( "Richard Mayhew", "richard" ) );
 	}
 
-	private void assertSearchResult(String expectedResult, List<RegisteredMember> results) {
+	private void assertAtLeastOneEntity(String expectedResult, String searchString) {
+		List<RegisteredMember> results = memberRegistration.search( searchString );
 		assertEquals( "Unexpected number of results from search", 1, results.size() );
```

Comment:
Right, I got confused with the other "assert" method (which I didn't implement right, either. I fixed both methods.